### PR TITLE
next.js: add option to use bundled runtime

### PR DESCRIPTION
**Pull request by feedthejim**
This PR adds an environment variable that should allow us to test the bundled version for Next.js on Vercel, see https://github.com/vercel/next.js/pull/52997 for reference.

The changes include:
- a new environment variable `VERCEL_NEXT_BUNDLED_SERVER`
- some logic changes that will put app route handlers into their own lambda groups
- extra logic to require early the rendering runtimes (see PR above for details)

### CODE REVIEW

File: .changeset/old-teachers-dress.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@vercel/next": patch
+"api": patch
```
Comment (styfle) [inline .changeset/old-teachers-dress.md]:
```suggestion
```

Remove `api` here since its not a public package

File: .changeset/poor-baboons-report.md
Comment (styfle) [inline .changeset/poor-baboons-report.md:1]: This whole file should be removed. Each PR should have a single changeset

File: packages/next/src/server-build.ts [hunk 2]
```diff
@@ -428,7 +451,11 @@ export async function serverBuild({
       }
 
       if (lambdaAppPaths[page]) {
-        appRouterPages.push(page);
+        if (lambdaAppPaths[page].fsPath.endsWith('route.js')) {
+          appRouteHandlers.push(page);
+        } else {
+          appRouterPages.push(page);
+        }
```
Comment (styfle) [inline packages/next/src/server-build.ts:458]: This change is not behind a flag. Is that intentional?

Comment (feedthejim) [reply packages/next/src/server-build.ts:458]: yes, we should have app route handlers into their own lambdas!

File: packages/next/test/fixtures/37-bundled-server/next.config.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  experimental: {
+    
+  },
+};
```
Comment (styfle) [inline packages/next/test/fixtures/37-bundled-server/next.config.js]: You can probably remove this file

File: packages/next/test/fixtures/37-bundled-server/probes.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    {
+      "path": "/api/hello",
+      "status": 200,
+      "mustContain": "hello world"
+    },
+    {
+      "path": "/",
+      "status": 200,
+      "mustContain": "hello world"
```
Comment (styfle) [inline packages/next/test/fixtures/37-bundled-server/probes.json]: All of these tests are looking for the same text. Can we update each so they return something different that that will ensure it works as expected?

**Review by styfle — APPROVED** [review]
Thanks!

Might be good to have @ijjk take a look too

**Review by ijjk — APPROVED** [review]

**Pull request merged**